### ISSUE TITLE
How to replicate Guanaco(OASST1) MMLU test results?

### ISSUE THREAD

**Issue by hangxu0304** [artidoro/qlora#194]
Hi, has anyone tried to replicate the MMLU test results of Guanaco (OASST1) in Table 5 in the paper? 
<img width="300" alt="image" src="https://github.com/artidoro/qlora/assets/52653060/6711e57e-6e7a-4ed9-b408-39095f5ad34f">

I have tried with the original training scripts provided at `./scripts/finetune_guanaco*.sh`. It turns out that the MMLU test score varies a lot with different dataset/training random seeds (see results below). The best scores I got for 7b and 65b Guanaco models are 37.4% and 61.4%, respectively.
<img width="797" alt="image" src="https://github.com/artidoro/qlora/assets/52653060/f5df62fc-ba55-4d5e-ac5f-d0b1a7b8f5f0">

Can any author clarify whether this matches the observation in your experiment, and how are the numbers reported in Table 5, e.g. using avg or max score? Thank you very much!

**Comment by artidoro** [comment]
Thanks for your question!

We used the original MMLU script to compute MMLU on the test set at the end of training by reloading the last model checkpoint. One thing to note is that in our finetuning script, we track dev MMLU during training which fluctuates more than on the test MMLU (the dev set is smaller than the test set). It is true that even the test MMLU has some variability. However, we only fine-tuned the Guanaco models once and did not compute statistics around MMLU test results.

Note that reloading the Guanaco checkpoints you trained might not work well due to the issue we patched #214. Sorry for the inconvenience!

**Comment by StiphyJay** [comment]
Hi, 

> Hi, has anyone tried to replicate the MMLU test results of Guanaco (OASST1) in Table 5 in the paper? <img alt="image" width="300" src="https://user-images.githubusercontent.com/52653060/250328618-6711e57e-6e7a-4ed9-b408-39095f5ad34f.png">
> 
> I have tried with the original training scripts provided at `./scripts/finetune_guanaco*.sh`. It turns out that the MMLU test score varies a lot with different dataset/training random seeds (see results below). The best scores I got for 7b and 65b Guanaco models are 37.4% and 61.4%, respectively. <img alt="image" width="797" src="https://user-images.githubusercontent.com/52653060/250330838-f5df62fc-ba55-4d5e-ac5f-d0b1a7b8f5f0.png">
> 
> Can any author clarify whether this matches the observation in your experiment, and how are the numbers reported in Table 5, e.g. using avg or max score? Thank you very much!

Hi. So how do we reproduce the reported results from paper. Evaluation again based on which script?

**Comment by Edenzzzz** [comment]
I took the hyperparams from the paper but got only 32.1 MMLU acc. Anyone figured out how to reproduce?
```
python qlora.py \
    --model_name_or_path huggyllama/llama-7b \
    --use_auth \
    --output_dir /fly/results/qlora \
    --logging_steps 10 \
    --save_strategy steps \
    --data_seed 42 \
    --save_steps 500 \
    --save_total_limit 40 \
    --evaluation_strategy steps \
    --eval_dataset_size 1024 \
    --max_eval_samples 1000 \
    --per_device_eval_batch_size 1 \
    --max_new_tokens 32 \
    --dataloader_num_workers 1 \
    --group_by_length \
    --logging_strategy steps \
    --remove_unused_columns False \
    --do_train \
    --do_eval \
    --do_mmlu_eval \
    --lora_r 64 \
    --lora_alpha 16 \
    --lora_modules all \
    --double_quant \
    --quant_type nf4 \
    --bf16 \
    --bits 16 \
    --warmup_ratio 0.03 \
    --lr_scheduler_type constant \
    --gradient_checkpointing \
    --dataset alpaca \
    --source_max_len 16 \
    --target_max_len 512 \
    --per_device_train_batch_size 1 \
    --gradient_accumulation_steps 16 \
    --max_steps 1875 \
    --eval_steps 187 \
    --learning_rate 0.0002 \
    --adam_beta2 0.999 \
    --max_grad_norm 0.3 \
    --lora_dropout 0.1 \
    --weight_decay 0.0 \
    --seed 0 \
    --mmlu_split test
```
